File: 洛谷题单/入门4_数组/P1047_校门外的树/solution.cpp
#include <iostream>
using namespace std;

const int MAXN = 10010;
// 这里只会让第一个元素变成 true,其他的还是false,单独拿出来展示一下
bool tree[MAXN] = {true};

int main() {
    int l, m;
    cin >> l >> m;
    // bool数组全局定义时默认为false,需要手动初始化
    for (int i = 0; i <= l; ++i)
        tree[i] = true;

    for (int i = 0; i < m; ++i) {
        int u, v;
        cin >> u >> v;
        for (int j = u; j <= v; ++j) {
            tree[j] = false;
        }
    }

    int cnt = 0;
    for (int i = 0; i <= l; ++i) {
        if (tree[i]) cnt++;
    }

    cout << cnt << '\n';
    return 0;
}

File: 洛谷题单/入门4_数组/P1047_校门外的树/solution.java
import java.util.Scanner;

public class Main {
    static final int MAXN = 10010;
    static boolean[] tree = new boolean[MAXN];

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int l = sc.nextInt();
        int m = sc.nextInt();

        for (int i = 0; i <= l; i++)
            tree[i] = true;

        for (int i = 0; i < m; i++) {
            int u = sc.nextInt();
            int v = sc.nextInt();
            for (int j = u; j <= v; j++)
                tree[j] = false;
        }

        int cnt = 0;
        for (int i = 0; i <= l; i++) {
            if (tree[i]) cnt++;
        }

        System.out.println(cnt);
    }
}
